### PR TITLE
WIP: Scala.js upgrade to v1.0.0 (#2392)

### PR DESCRIPTION
**Pull request by lvitaly**

### CODE REVIEW

**Comment by flavomulti** [comment]
I can't quite tell what the errors are, how could they fail on user input?

**Comment by jdegoes** [comment]
@lvitaly Thank you for submitting your first pull request!!

Can you merge conflicts so we can see if this passes now? Thank you! 🙏

**Comment by lvitaly** [comment]
@jdegoes Thanks for your welcoming!
I merged the latest changes from master and all looks fine except magnolia tests. There is no build of magnolia for ScalaJS 1.0.0-RC2 (https://mvnrepository.com/artifact/com.propensive/magnolia).

**Comment by jdegoes** [comment]
@lvitaly Can you open up a ticket on Magnolia for adding support for 1.0.0-RC2? Would be nice to get that released before we merge this one so we don't have to disable any tests (if possible)...

**Comment by lvitaly** [comment]
@jdegoes I've created the issue https://github.com/propensive/magnolia/issues/205. I hope it will be available soon.

**Comment by jdegoes** [comment]
@lvitaly Good news! Magnolia now supports Scala.js 1.0. 

So this is ready to push through if you have the time!

**Comment by adamgfraser** [comment]
@lvitaly Can't we upgrade to ScalaJS 1.0.0 now that it has been released?

@neko-kai @pshirshov Would it be possible to publish `reflect` for ScalaJS 1.0.0?

**Comment by mijicd** [comment]
@adamgfraser I think there will be problems with `scala-java-time` as it isn't published for `1.0.0`.

**Comment by lvitaly** [comment]
@jdegoes Thanks for the update! Magnolia supports Scala.js 1.0 but the artifact is not published yet 😔 https://mvnrepository.com/artifact/com.propensive/magnolia
And there is another issue with `scala-java-time` https://github.com/cquiroz/scala-java-time/issues/154
@adamgfraser I've merged latest master and going to upgrade to 1.0.0

**Comment by adamgfraser** [comment]
@mijicd You are totally right.

@lvitaly Alright, sounds like we are in a holding pattern here. Thanks for your work on this!

**Comment by lvitaly** [comment]
@adamgfraser @neko-kai @pshirshov I created a pull request for updating `izumi-reflect` to ScalaJS 1.0.0
https://github.com/zio/izumi-reflect/pull/3

**Comment by neko-kai** [comment]
That PR passes CI now https://github.com/zio/izumi-reflect/pull/3
@adamgfraser you can merge it when it's time to publish for 1.0.0. I guess RC18 will still be for 0.6 due to `scala-java-time`?

**Comment by neko-kai** [comment]
Given  https://github.com/cquiroz/scala-java-time/issues/154
> This is blocked by a bug on Scala.js which should be fixed on 1.0.1

We may be blocked till 1.0.1 on `scala-java-time`

**Comment by aappddeevv** [comment]
Looks like 1.0.1 is out. Hopefully cquiroz can release his update that would help unblock this.

**Comment by ghostdogpr** [comment]
scala-java-time was published: https://github.com/cquiroz/scala-java-time/blob/master/CHANGELOG.md

**Comment by jdegoes** [comment]
@lvitaly With Scala Java Time released, anything else we're blocking on?

**Comment by neko-kai** [comment]
@lvitaly 
`izumi-reflect` for SJS 1.0.0 is out https://github.com/zio/izumi-reflect/releases/tag/v0.12.0-M1

**Comment by lvitaly** [comment]
@neko-kai, thank you for your help.
@jdegoes, looks like we are almost ready and just waiting for the appearance `izumi-reflect` in the maven repo.
https://mvnrepository.com/artifact/dev.zio/izumi-reflect

**Comment by neko-kai** [comment]
It's up on sonatype now https://repo1.maven.org/maven2/dev/zio/izumi-reflect_2.13/0.12.0-M1/

File: build.sbt [hunk 1]
```diff
@@ -270,7 +270,7 @@ lazy val testRunner = crossProject(JVMPlatform, JSPlatform)
     ),
     mainClass in (Test, run) := Some("zio.test.sbt.TestMain")
   )
-  .jsSettings(libraryDependencies ++= Seq("org.scala-js" %% "scalajs-test-interface" % "0.6.32"))
+  .jsSettings(libraryDependencies ++= Seq("org.scala-js" %% "scalajs-test-interface" % "1.0.0-RC2"))
```
Comment (ghostdogpr) [inline build.sbt]: There’s a 1.0.1 version now.

**Comment by ghostdogpr** [comment]
Magnolia 0.12.8 is available with ScalaJS 1.0

**Comment by ghostdogpr** [comment]
Could we either cross-publish for 0.6 and 1.0 (like cats and cats-effect) or wait until after a first ZIO 1.0.0 to get this in? Otherwise projects that are stuck to 0.6 (because of some other dependencies) won't be able to upgrade ZIO.

File: build.sbt (hunk 1 above)
Comment (lvitaly) [reply build.sbt]: @ghostdogpr thanks, updated

**Comment by neko-kai** [comment]
@ghostdogpr how about 18-2 with 0.6, then immediately merge this and release 18-3 with 1.0? I think supporting 0.6 is nonviable long-term, everyone should jump ship

**Comment by ghostdogpr** [comment]
Fair enough. Hopefully monix, fs2, sttp etc will be available for ScalaJS 1.x soon...

**Comment by ghostdogpr** [comment]
Actually, how about we wait until ZIO 1.0.0 and only then, we publish with ScalaJS 1.0 right after?

Otherwise, a lot of projects will be stuck with an old binary incompatible ZIO and impossible to upgrade. For example let's take interop-monix. Monix doesn't have a version for ScalaJS 1.x yet. So we won't be able to upgrade ZIO in that project anymore. Same with sttp, caliban, etc. If we wait ZIO 1.0 it won't be a problem because of binary compatibility: the library will stay on ZIO 1.0.0 but library users will be able to use any ZIO 1.x since it will be binary compatible.

**Comment by neko-kai** [comment]
Makes sense. Although if 1.0 takes longer, we might have to bite the bullet and cross-publish

**Comment by ghostdogpr** [comment]
Since I wrote that, support was added in many libraries. fs2 supports it, monix just merged support a few days ago (not released yet), etc. So, I think it's okay if we merge that (we should probably rebase and upgrade to 1.0.1 before).

**Comment by neko-kai** [comment]
Let's merge this one and upgrade to 1.0.1 in a separate PR

**Pull request merged**